java-client/build.gradle.kts

```diff
@@ -348,40 +351,43 @@ publishing {
 }
 
 if (runtimeJavaVersion >= JavaVersion.VERSION_11) {
-  val java11: SourceSet = sourceSets.create("java11") {
-    java {
-      compileClasspath += sourceSets.main.get().output + sourceSets.test.get().output
-      runtimeClasspath += sourceSets.main.get().output + sourceSets.test.get().output
-      srcDir("src/test/java11")
+    val java11: SourceSet = sourceSets.create("java11") {
+        java {
+            compileClasspath += sourceSets.main.get().output + sourceSets.test.get().output
+            runtimeClasspath += sourceSets.main.get().output + sourceSets.test.get().output
+            srcDir("src/test/java11")
+        }
     }
-  }
 
-  configurations[java11.implementationConfigurationName].extendsFrom(configurations.testImplementation.get())
-  configurations[java11.runtimeOnlyConfigurationName].extendsFrom(configurations.testRuntimeOnly.get())
+    configurations[java11.implementationConfigurationName].extendsFrom(configurations.testImplementation.get())
+    configurations[java11.runtimeOnlyConfigurationName].extendsFrom(configurations.testRuntimeOnly.get())
+
+    dependencies {
+        testImplementation("org.opensearch.test", "framework", opensearchVersion) {
+            exclude(group = "org.hamcrest")
+        }
+
+        // Apache 2.0
+        testImplementation("org.wiremock", "wiremock", "3.9.2")
```

Comment:
Nothing against `wiremock`, but we already have all needed pieces:
 - `HttpServer` that comes from HttpCore 5 (example https://github.com/apache/httpcomponents-core/blob/master/httpcore5-reactive/src/test/java/org/apache/hc/core5/reactive/examples/ReactiveFullDuplexServerExample.java)
 - since we run test with JDK-11, we could use `com.sun.net.httpserver.HttpServer` that comes with standard library (see please https://docs.oracle.com/en/java/javase/11/docs/api/jdk.httpserver/com/sun/net/httpserver/HttpServer.html)

Reply:
Have dropped wiremock and re-implemented the necessary features